### PR TITLE
Move NodeLabel priority logic to its Score and Filter plugin

### LINKED ISSUE #86437: Move NodeLabelPriority to its Score and Filter plugin [Migration Phase 2]
/sig scheduling
/priority important-soon

See #86399 as an example.

Part of #85822

This shouldn't need a PostFilter Plugin.

/assign @danielqsj

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

/kind cleanup

**Which issue(s) this PR fixes**:
Fixes #86437

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 1]
```diff
@@ -71,19 +71,18 @@ func New(plArgs *runtime.Unknown, handle framework.FrameworkHandle) (framework.P
 		return nil, err
 	}
 	// Note that the reduce function is always nil therefore it's ignored.
```
Comment (liu-cong) [inline pkg/scheduler/framework/plugins/nodelabel/node_label.go]: Delete comment.

**Review by liu-cong — COMMENTED** [review]
Minor comment otherwise lgtm.

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/nodelabel/node_label.go]: done.

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 2]
```diff
@@ -93,10 +91,31 @@ func (pl *NodeLabel) Name() string {
...
+				return false
+			}
+		}
+		return true
+	}
+	if check(pl.PresentLabels, true) && check(pl.AbsentLabels, false) {
+		return nil
+	}
+
+	return migration.PredicateResultToFrameworkStatus([]predicates.PredicateFailureReason{predicates.ErrNodeLabelPresenceViolated}, nil)
```
Comment (liu-cong) [inline pkg/scheduler/framework/plugins/nodelabel/node_label.go]: Can we return the status directly instead of doing the translation?

Comment (ahg-g) [reply pkg/scheduler/framework/plugins/nodelabel/node_label.go]: I want to keep the failure reason the same and deal with moving those later, there are some dependencies on them.

Comment (Huang-Wei) [reply pkg/scheduler/framework/plugins/nodelabel/node_label.go]: @liu-cong I'm working on a PR handling all the cases.

**Pull request merged**